File: apps/api/app/schemas/asset.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


AssetType = Literal["image", "video", "audio", "reference"]


class AssetBase(BaseModel):
    type: AssetType
    role: str
    relative_path: str = Field(min_length=1)
    thumbnail_path: str | None = None
    mime_type: str = Field(min_length=1)
    width: int | None = Field(default=None, gt=0)
    height: int | None = Field(default=None, gt=0)
    duration_seconds: float | None = Field(default=None, ge=0)
    size_bytes: int = Field(ge=0)
    hash: str | None = None


class AssetCreate(AssetBase):
    project_id: str
    scene_id: str | None = None


class AssetRead(AssetBase):
    model_config = ConfigDict(from_attributes=True)

    id: str
    project_id: str
    scene_id: str | None
    created_at: datetime

File: apps/web/src/types/asset.ts
export type AssetType = 'image' | 'video' | 'audio' | 'reference'

export type Asset = {
  id: string
  project_id: string
  scene_id: string | null
  type: AssetType
  role: string
  relative_path: string
  thumbnail_path: string | null
  mime_type: string
  width: number | null
  height: number | null
  duration_seconds: number | null
  size_bytes: number
  hash: string | null
  created_at: string
}